app/cards/create/page.tsx

```diff
@@ -25,38 +28,42 @@ export default function Cards() {
       dateOfBirth: Yup.string().required("Обов'язкове поле"),
       gender: Yup.string().required("Обов'язкове поле"),
     }),
-    onSubmit: (values) => {
-      const newCard = createCard(
-        values.name.toUpperCase(),
-        values.dateOfBirth,
-        values.gender,
-      )
-
-      router.push(`/cards/${newCard.id}`)
+    onSubmit: async (values) => {
+      try {
+        const cardData = {
+          name: values.name.toUpperCase(),
+          dateOfBirth: values.dateOfBirth,
+          gender: values.gender,
+        }
+        const userId = user.id
+        const newCard = await createCard(userId, cardData)
+        router.push(`/cards/${newCard.id}`)
+      } catch (error) {
+        console.error("Помилка при створенні карти:", error)
+      }
     },
   })
   const showSuccessAlert = () => toast.success("Карточку створено!")
-  const { getFieldProps } = formik
+
+  const { getFieldProps, errors, touched } = formik
+  const getErrorMessage = (key: keyof typeof formik.touched) =>
+    touched[key] && errors[key] ? errors[key] : null
+
   return (
     <div className="mt-10 text-center">
       <h2 className="text-grey font-bold">Внесіть дані Вашої дитини</h2>
 
       <form className="flex-col mt-5  " onSubmit={formik.handleSubmit}>
         <Input {...getFieldProps("name")} label="Ім'я" type="text" />
-        {formik.touched.name && formik.errors.name ? (
-          <div className="text-red-500 text-xs">{formik.errors.name}</div>
-        ) : null}
+        <ErrorMessage>{getErrorMessage("name")}</ErrorMessage>
```

Comment:
You have an error inside `Input` component, so you can do like this:

```tsx
<Input {...getFieldProps("name")} label="Ім'я" type="text" error={getErrorMessage("name")} />
```

Please, change it in all places.

Also in Input change: 
`{error ? <p className="text-red-500 text-xs">{error}</p> : null}` to `<ErrorMessage>{error}</ErrorMessage>`